ghcide/src/Development/IDE/Core/FileExists.hs

```diff
@@ -119,10 +120,10 @@ modifyFileExists state changes = do
     -- flush previous values
     let (fileModifChanges, fileExistChanges) =
             partition ((== FileChangeType_Changed) . snd) changes
-    mapM_ (deleteValue (shakeExtras state) GetFileExists . fst) fileExistChanges
-    io1 <- recordDirtyKeys (shakeExtras state) GetFileExists $ map fst fileExistChanges
-    io2 <- recordDirtyKeys (shakeExtras state) GetModificationTime $ map fst fileModifChanges
-    return (io1 <> io2)
+    keys0 <- concat <$> mapM (deleteValue (shakeExtras state) GetFileExists . fst) fileExistChanges
```

Comment:
Maybe slightly controversial tip (up to you if you like it more)
but you can use this instance here `Monoid a => Monoid (STM a)`:
```suggestion
    keys0 <- foldMap (deleteValue (shakeExtras state) GetFileExists . fst) fileExistChanges
```

Reply:
Sorry for bad tip - it looks like this instance was introduces with base coming with ghc 9.4 and is not yet available with ghc 9.2 :/

Reply:
right, I discovered that before, STM do not have the monoid instance, hit by it again lol